contracts/stellar-interchain-token-service/src/token_handler.rs

```diff
@@ -38,16 +39,25 @@ pub fn give_token(
     }: TokenIdConfigValue,
     amount: i128,
 ) -> Result<(), ContractError> {
-    let token_manager = TokenManagerClient::new(env, &token_manager);
+    let token_manager_client = TokenManagerClient::new(env, &token_manager);
 
     match token_manager_type {
         TokenManagerType::NativeInterchainToken => {
-            token_manager.mint(env, &token_address, recipient, amount)
+            token_manager_client.mint_from(env, &token_address, recipient, amount)
         }
         TokenManagerType::LockUnlock => {
-            token_manager.transfer(env, &token_address, recipient, amount)
+            token_manager_client.transfer(env, &token_address, recipient, amount)
+        }
+        TokenManagerType::MintBurn => {
+            if InterchainTokenClient::new(env, &token_address)
+                .try_token_id()
+                .is_ok()
+            {
+                token_manager_client.mint_from(env, &token_address, recipient, amount);
+            } else {
+                token_manager_client.mint(env, &token_address, recipient, amount);
+            }
```

Comment:
interchain tokens aren't meant to use mint/burn type. We should always call `mint` here. Also, this is not reliable. A token may implement `token_id` method but might not actually be an interchain token.

To facilitate testing, create a test/example custom token that implements this interface instead